### PR TITLE
migrate proxy/iptables/proxier.go logs to structured logging

### PR DESCRIPTION
**Pull request by JornShen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

part  of https://github.com/kubernetes/enhancements/issues/1602

**Which issue(s) this PR fixes**:

[keps/sig-instrumentation/1602-structured-logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)

[Structured Logging migration instructions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @thockin 
let me add Tim to the PR, I'm not familiar enough with the KEP for structured logging

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -277,13 +277,13 @@ func NewProxier(ipt utiliptables.Interface,
...
 	if val, err := sysctl.GetSysctl(sysctlBridgeCallIPTables); err == nil && val != 1 {
-		klog.Warning("missing br-netfilter module or unset sysctl br-nf-call-iptables; proxy may not work as intended")
+		klog.InfoS("missing br-netfilter module or unset sysctl br-nf-call-iptables; proxy may not work as intended")
 	}
 
 	// Generate the masquerade mark to use for SNAT rules.
 	masqueradeValue := 1 << uint(masqueradeBit)
 	masqueradeMark := fmt.Sprintf("%#08x", masqueradeValue)
-	klog.V(2).Infof("iptables(%s) masquerade mark: %s", ipt.Protocol(), masqueradeMark)
+	klog.V(2).InfoS("generate iptables masquerade", "protocol", ipt.Protocol(), "mark", masqueradeMark)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:286]: We should use the word "family" or "ipFamily" rather than "protocol" here to be consistent

Comment (thockin) [reply pkg/proxy/iptables/proxier.go:286]: I think the message should be "using iptables mark for masquerade", with key "mark"

File: pkg/proxy/iptables/proxier.go [hunk 2]
```diff
@@ -298,7 +298,7 @@ func NewProxier(ipt utiliptables.Interface,
 	nodePortAddresses = ipFamilyMap[ipFamily]
 	// Log the IPs not matching the ipFamily
 	if ips, ok := ipFamilyMap[utilproxy.OtherIPFamily(ipFamily)]; ok && len(ips) > 0 {
-		klog.Warningf("IP Family: %s, NodePortAddresses of wrong family; %s", ipFamily, strings.Join(ips, ","))
+		klog.InfoS("IPs not matching the ipFamily", "IP Family", ipFamily, "NodePortAddresses", strings.Join(ips, ","))
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:301]: "found node IPs of the wrong family", with keys "ipFamily" or "family" and "ips"

File: pkg/proxy/iptables/proxier.go [hunk 3]
```diff
@@ -331,8 +331,7 @@ func NewProxier(ipt utiliptables.Interface,
 	}
 
 	burstSyncs := 2
-	klog.V(2).Infof("iptables(%s) sync params: minSyncPeriod=%v, syncPeriod=%v, burstSyncs=%d",
-		ipt.Protocol(), minSyncPeriod, syncPeriod, burstSyncs)
+	klog.V(2).InfoS("iptables sync params", "protocol", ipt.Protocol(), "minSyncPeriod", minSyncPeriod, "syncPeriod", syncPeriod, "burstSyncs", burstSyncs)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:334]: s/protocol/family as elsewhere

File: pkg/proxy/iptables/proxier.go [hunk 4]
```diff
@@ -343,9 +342,9 @@ func NewProxier(ipt utiliptables.Interface,
 		proxier.syncProxyRules, syncPeriod, wait.NeverStop)
 
 	if ipt.HasRandomFully() {
-		klog.V(2).Infof("iptables(%s) supports --random-fully", ipt.Protocol())
+		klog.V(2).InfoS("iptables supports --random-fully", "protocol", ipt.Protocol())
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:345]: s/protocol/family

File: pkg/proxy/iptables/proxier.go [hunk 5]
```diff
@@ -663,7 +662,8 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 // is observed.
 func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
 	if node.Name != proxier.hostname {
-		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		klog.ErrorS(nil, "Received a watch event for a node that doesn't match the current node",
+			"watch node", node.Name, "current node", proxier.hostname)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:666]: "eventNode" and "currentNode" ?

File: pkg/proxy/iptables/proxier.go [hunk 6]
```diff
@@ -682,7 +682,8 @@ func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
 // node object is observed.
 func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
 	if node.Name != proxier.hostname {
-		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		klog.ErrorS(nil, "Received a watch event for a node that doesn't match the current node",
+			"watch node", node.Name, "current node", proxier.hostname)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:686]: same as above

File: pkg/proxy/iptables/proxier.go [hunk 7]
```diff
@@ -701,7 +702,8 @@ func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
 // object is observed.
 func (proxier *Proxier) OnNodeDelete(node *v1.Node) {
 	if node.Name != proxier.hostname {
-		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		klog.ErrorS(nil, "Received a watch event for a node that doesn't match the current node",
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:705]: same

File: pkg/proxy/iptables/proxier.go [hunk 8]
```diff
@@ -815,30 +817,30 @@ func (proxier *Proxier) syncProxyRules() {
...
+		klog.V(2).InfoS("Not syncing iptables until Services and Endpoints have been received from master")
 		return
 	}
 
 	// Keep track of how long syncs take.
 	start := time.Now()
 	defer func() {
 		metrics.SyncProxyRulesLatency.Observe(metrics.SinceInSeconds(start))
-		klog.V(2).Infof("syncProxyRules took %v", time.Since(start))
+		klog.V(2).InfoS("syncProxyRules", "tookTime", time.Since(start))
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:828]: How about "syncProxyRules complete" and key "elapsed"

File: pkg/proxy/iptables/proxier.go [hunk 9]
```diff
@@ -815,30 +817,30 @@ func (proxier *Proxier) syncProxyRules() {
...
+		klog.InfoS("No local addresses found, assuming all external IPs are not local")
 	}
 
 	localAddrSet := utilnet.IPSet{}
 	localAddrSet.Insert(localAddrs...)
 
 	nodeAddresses, err := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
 	if err != nil {
-		klog.Errorf("Failed to get node ip address matching nodeport cidrs %v, services with nodeport may not work as intended: %v", proxier.nodePortAddresses, err)
+		klog.ErrorS(err, "Failed to get node ip address matching nodeport cidrs, services with nodeport may not work as intended", "cidrs", proxier.nodePortAddresses)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:843]: in the message it should be CIDRs :)

File: pkg/proxy/iptables/proxier.go [hunk 10]
```diff
@@ -851,44 +853,44 @@ func (proxier *Proxier) syncProxyRules() {
 	// merge stale services gathered from updateEndpointsMap
 	for _, svcPortName := range endpointUpdateResult.StaleServiceNames {
 		if svcInfo, ok := proxier.serviceMap[svcPortName]; ok && svcInfo != nil && conntrack.IsClearConntrackNeeded(svcInfo.Protocol()) {
-			klog.V(2).Infof("Stale %s service %v -> %s", strings.ToLower(string(svcInfo.Protocol())), svcPortName, svcInfo.ClusterIP().String())
+			klog.V(2).InfoS("Stale service", "protocol", strings.ToLower(string(svcInfo.Protocol())), "svcPortName", svcPortName.String(), "clusterIP", svcInfo.ClusterIP().String())
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:856]: protocol here is correct

File: pkg/proxy/iptables/proxier.go [hunk 11]
```diff
@@ -1526,7 +1528,7 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 		// Ignore IP addresses with incorrect version
 		if isIPv6 && !utilnet.IsIPv6String(address) || !isIPv6 && utilnet.IsIPv6String(address) {
-			klog.Errorf("IP address %s has incorrect IP version", address)
+			klog.ErrorS(nil, "IP has incorrect IP version", "address", address)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:1531]: key "ip"

File: pkg/proxy/iptables/proxier.go [hunk 12]
```diff
@@ -1604,7 +1606,7 @@ func (proxier *Proxier) syncProxyRules() {
 		for _, lastChangeTriggerTime := range lastChangeTriggerTimes {
 			latency := metrics.SinceInSeconds(lastChangeTriggerTime)
 			metrics.NetworkProgrammingLatency.Observe(latency)
-			klog.V(4).Infof("Network programming of %s took %f seconds", name, latency)
+			klog.V(4).InfoS("Network programming", "endpoint", klog.KRef(name.Namespace, name.Name), "tookTime", latency)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:1609]: s/tookTime/elapsed ?

File: pkg/proxy/iptables/proxier.go [hunk 13]
```diff
@@ -1625,21 +1627,21 @@ func (proxier *Proxier) syncProxyRules() {
...
 	}
 	if err := proxier.serviceHealthServer.SyncEndpoints(endpointUpdateResult.HCEndpointsLocalIPSize); err != nil {
-		klog.Errorf("Error syncing healthcheck endpoints: %v", err)
+		klog.ErrorS(err, "Error syncing healthcheck endpoints")
 	}
 
 	// Finish housekeeping.
 	// TODO: these could be made more consistent.
-	klog.V(4).Infof("Deleting stale services IPs: %v", staleServices.UnsortedList())
+	klog.V(4).InfoS("Deleting stale services", " IPs", staleServices.UnsortedList())
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:1638]: s/ IPs/ips  (Note the space)

File: pkg/proxy/iptables/proxier.go [hunk 14]
```diff
@@ -1625,21 +1627,21 @@ func (proxier *Proxier) syncProxyRules() {
...
 	for _, svcIP := range staleServices.UnsortedList() {
 		if err := conntrack.ClearEntriesForIP(proxier.exec, svcIP, v1.ProtocolUDP); err != nil {
-			klog.Errorf("Failed to delete stale service IP %s connections, error: %v", svcIP, err)
+			klog.ErrorS(err, "Failed to delete stale service connections", "IP", svcIP)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:1641]: ip

File: pkg/proxy/iptables/proxier.go [hunk 15]
```diff
@@ -1625,21 +1627,21 @@ func (proxier *Proxier) syncProxyRules() {
...
 		}
 	}
-	klog.V(4).Infof("Deleting stale endpoint connections: %v", endpointUpdateResult.StaleEndpoints)
+	klog.V(4).InfoS("Deleting stale endpoint connections", "endpoint", endpointUpdateResult.StaleEndpoints)
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go:1644]: This is plural, so "endpoints" ?

**Review by thockin — COMMENTED** [review]
Do we have a convention for key names?  I guess they should be named in the usual goVariableName syntax?

File: pkg/proxy/iptables/proxier.go (hunk 8 above)
Comment (JornShen) [reply pkg/proxy/iptables/proxier.go:828]: good!

**Comment by JornShen** [comment]
@thockin  thanks for review! 
 updated.  PTAL :)

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**